src/code/client.coffee

```diff
@@ -136,6 +136,9 @@ class CloudFileManagerClient
     @newFileOpensInNewTab = if @appOptions.ui?.hasOwnProperty('newFileOpensInNewTab') then @appOptions.ui.newFileOpensInNewTab else true
     @newFileAddsNewToQuery = @appOptions.ui?.newFileAddsNewToQuery
 
+    if @appOptions.ui?.confirmCloseIfDirty
+      @_setupConfirmOnClose()
+
```

Comment:
CODAP already confirms close on dirty documents. It must use a different mechanism. I realize this feature is enabled with a flag so there isn't a danger of conflict, but should there be two mechanisms to accomplish the same thing?

Reply:
Looking at the CODAP implementation, it appears that the CODAP requirements may be more elaborate than the CFM implementation, so there may be a reason to have two implementations.
```
if (DG.currDocumentController().get('hasUnsavedChanges') &&
    (DG.get('embeddedMode') === 'no') &&
    (DG.get('componentMode') === 'no')) {
  return 'DG.AppController.beforeUnload.confirmationMessage'.loc();
}
```